content/articles/how-to-create-a-machine-learning-app-using-the-fastapi-and-deploying-it-to-the-kubernetes-cluster/index.md

```diff
@@ -0,0 +1,564 @@
+
+We build our model using the [Naive Bayes Classifier](https://en.wikipedia.org/wiki/Naive_Bayes_classifier).
+Naive Bayes is a simple classification algorithm to train and fit our model.
+
+```python
+nb = MultinomialNB()
+nb.fit(x_train,y_train)
+```
+
+### Checking the accuracy of our model
+
+We have to check the accuracy score of our model to know how well we trained our model.
+
+```python
+nb.score(x_test,y_test)
+```
+
+### Making predictions
+
+After we have trained our model, we can now feed our model with new inputs to start making predictions.
+Our model will make accurate predictions based on how well we trained it. Therefore, the higher the accuracy score, the better our model will be in making predictions.
+
+```python
+name1 = ["Yin","Bathsheba","Brittany","Vladmir"]
+
+vector1 = vec.transform(sample1).toarray()
+
+nb.predict(vector1)
+```
+
+### Saving our model using joblib
+
+We will use joblib to save our model into a pickle file.
+Picking our model makes it easier to use our model in the future without repeating the training process.
+A pickle file is a byte stream of our model, by importing joblib from sklearn
+
+```python
+import joblib from sklearn.externals
+```
+
+```python
+nationality_predictor = open("naive_bayes.pkl","wb")
+joblib.dump(cv,nationality_vectorizer)
+nationality_predictor.close()
+```
+
+We will name our pickle file 'naive_bayes.pkl'.
+
+### Introduction to the FastAPI
+
+[FastAPI](https://fastapi.tiangolo.com/) is a new python-based web framework used to create Web APIs. FastAPi is fast when serving the application; this enhances the app's performance.
+It is used with Python 3.6+.
+Fast API is potent since it combines the functionalities of best frameworks such as [flask](https://flask.palletsprojects.com/) and [swagger](https://swagger.io/)
+
+### Installing the FastApi
+
+Use the following commands:
+
+```python
+pip install fast API
+```
+
+Let's install the server.
+
+```python
+pip install uvicorn[standard]
+```
+
+### Creating and API for our machine learning model using the FastAPI
+
+1. First, create a new Python file and name it 'main.py.'
+2. Add our pickle file 'naive_bayes.pkl' in a new folder.
+
+The folder structure is as shown.
+
+```bash
+├── app.py
+├── model
+   ├── naive_bayes.pkl
+
+```
+
+- Let's get started working with our new 'app.py' file.
+
+### Importing our FastApi packages
+
+```python
+import uvicorn
+import FastAPI, Query from fast API
+```
+
+### Loading ML packages
+
+We will use joblib to unpickle our previously pickled file.
+
+```python
+import joblib from sklearn.externals
+```
+
+### Unplickling our naive Bayes classifier file
+
+```python
+nationality_naive_bayes = open("model/naive_bayes.pkl","rb")
+nationality_cv = joblib.load(nationality_naive_bayes)
+```
+
+### initializing our app
+
+```python
+app = FastAPI()
+```
+
+### Creating our routes
+
+We will create a simple route that will run localhost port 8000
+
+```python
+@app.get('/')
+async def index():
```

Comment:
Explain the code. async is an advanced concept and needs to be explained in depth